platform/pensando/sonic-platform-modules-dpu/dpu/utils/dpu_db_util.py

```diff
@@ -0,0 +1,602 @@
+                spec.EventFilter.Types.append(event)
+            yield grpcmsg
+
+    def _process_event(self, event):
+        global g_count
+
+        try:
+            event_type = event.EventInfo.Type
+            event_description = event.EventInfo.Description
+            event_message = event.EventInfo.Message
+        except Exception as e:
+            log_err(f"Failed to process event due to {e}")
+            return
+
+        try:
+            if event_type in CRITICAL_EVENTS:
+                # log event
+                g_count += 1
+                self._log_emerg(f"Event # {g_count} Time -> {time.asctime()}")
+                self._log_emerg(f"Received event {event}")
+
+                # update dpu state db for data plane
+                if self.db == None:
+                    return
+                dpu_data_plane_status = 'DOWN'
+                dpu_data_plane_reason = event_description
+                fvs_data_plane_data = [
+                    ('dpu_data_plane_state', dpu_data_plane_status),
+                    ('dpu_data_plane_time', datetime.now().strftime('%Y%m%d %H:%M:%S')),
+                    ('dpu_data_plane_reason', dpu_data_plane_reason)
+                ]
+                for name, value in fvs_data_plane_data:
+                    self.db.hset(self.table, name, value)
+        except Exception as e:
+            log_err(f"Failed to update dpu state db data plane fields due to {e}")
+
+        try:
+            if event_type in NETWORK_EVENTS:
+                # log event
+                g_count += 1
+                log_info(f"Event # {g_count} Time -> {time.asctime()}")
+                log_info(f"Received event {event}")
+
+                # update dpu state db for control plane
+                if self.db == None:
+                    return
+                self._update_dpu_control_plane_db()
+        except Exception as e:
+            log_err(f"Failed to update dpu state db control plane fields due to {e}")
+
+    def _event_listener(self):
+        channel_addr = "{}:{}".format(LOCALHOST,str(EVENT_PORT))
+        channel = grpc.insecure_channel(channel_addr)
+        stub = oper_pb2_grpc.OperSvcStub(channel)
+        resp = stub.OperInfoSubscribe(self._getGrpcEventMessage())
+        time.sleep(1)
+        for event in resp:
+            self._process_event(event)
+
+    def start(self):
+        # spawn operd listener thread
+        self.event_stop = False
+        if (self.event_thread == None) or (not self.event_thread.is_alive()):
+            self.event_thread = threading.Thread(target=self._event_listener)
+            self.event_thread.daemon = True
+            self.event_thread.start()
+
+    def stop(self):
+        self.log_warning("Stopping event listener thread")
+        self.event_stop = True
+        if self.event_thread is not None:
+            self.event_thread.join()
+            self.event_thread = None
+
+class RebootCauseUpdater():
+    def __init__(self, chassis, db):
```

Comment:
@SahilChaudhari As per latest design, NPU can record the DPU reboot please sync with Ramesh.

Reply:
We are pushing the latest code as per new design